### PR TITLE
Feature/eco 2666 add risk information

### PR DESCRIPTION
**Pull request by Ruslan05**
- Developer(s): @Ruslan05

- Ticket: https://spryker.atlassian.net/browse/ECO-2666

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Heidelpay             | minor                 |                       |

#### Release Notes

{skip}

-----------------------------------------

##### Change log

Improvements

- Added `HeidelpayRiskInformation` transfer.
- Added `riskInformation ` property  to `HeidelpayRequest ` transfer.
- Added sending of risk information during initialize request.
- Added `HeidelpayToSalesFacadeInterface.getCustomerOrders()` to dependencies.

### CODE REVIEW

File: src/SprykerEco/Zed/Heidelpay/Business/Adapter/Mapper/RequestToHeidelpay.php [hunk 1]
```diff
@@ -53,6 +53,12 @@ public function map(HeidelpayRequestTransfer $requestTransfer, Request $heidelpa
             $requestTransfer->getCustomerPurchase()->getSecret()
         );
 
+        $riskInformationTransfer = $requestTransfer->getRiskInformation();
+
+        $heidelpayRequest->getRiskInformation()->setCustomerSince($riskInformationTransfer->getCustomerSince());
+        $heidelpayRequest->getRiskInformation()->setCustomerGuestCheckout($riskInformationTransfer->getIsCustomerGuest());
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Adapter/Mapper/RequestToHeidelpay.php]: Shouldn't this be `setIsCustomerGuest`?

File: src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php [hunk 2]
```diff
@@ -82,9 +84,43 @@ protected function mapQuoteInformation(QuoteTransfer $quoteTransfer, HeidelpayRe
                 ->setIdOrder($this->generateQuoteId($quoteTransfer))
         );
 
+        $customerRegistrationDate = $this->findCustomerRegistrationDate($quoteTransfer->getCustomer());
+
+        $heidelpayRequestTransfer->setRiskInformation(
+            (new HeidelpayRiskInformationTransfer())
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php]: Mb you can move this to the separate method?

File: src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php [hunk 3]
```diff
@@ -82,9 +84,43 @@ protected function mapQuoteInformation(QuoteTransfer $quoteTransfer, HeidelpayRe
...
+    }
+
+    /**
+     * @param string $date
+     *
+     * @return string
+     */
+    protected function formatDate(string $date): string
+    {
+        $dateArray = explode(' ', $date);
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php]: I would prefer to have this in the proper way. E.g. smth like `(new \DateTime($date))->format('Y-m-d');`

File: src/SprykerEco/Zed/Heidelpay/Dependency/Facade/HeidelpayToSalesFacadeBridge.php [hunk 4]
```diff
@@ -31,4 +33,15 @@ public function getOrderByIdSalesOrder($idSalesOrder)
     {
         return $this->salesFacade->getOrderByIdSalesOrder($idSalesOrder);
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderListTransfer $orderListTransfer
+     * @param $idCustomer
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Dependency/Facade/HeidelpayToSalesFacadeBridge.php]: type

File: src/SprykerEco/Zed/Heidelpay/Dependency/Facade/HeidelpayToSalesFacadeBridge.php [hunk 5]
```diff
@@ -31,4 +33,15 @@ public function getOrderByIdSalesOrder($idSalesOrder)
...
+     *
+     * @return \Generated\Shared\Transfer\OrderListTransfer
+     */
+    public function getCustomerOrders(OrderListTransfer $orderListTransfer, $idCustomer)
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Dependency/Facade/HeidelpayToSalesFacadeBridge.php]: This should be described in the change log

File: src/SprykerEco/Zed/Heidelpay/Business/Adapter/Mapper/RequestToHeidelpay.php (hunk 1 above)
Comment (Ruslan05) [reply src/SprykerEco/Zed/Heidelpay/Business/Adapter/Mapper/RequestToHeidelpay.php]: This is comming from `Heidelpay` namespace:
https://github.com/heidelpay/php-payment-api/blob/master/lib/ParameterGroups/RiskInformationParameterGroup.php#L73

File: src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php (hunk 2 above)
Comment (Ruslan05) [reply src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php]: Moved to `mapCustoemrInformation()`.

File: src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php [hunk 6]
```diff
@@ -85,6 +88,47 @@ protected function mapQuoteInformation(QuoteTransfer $quoteTransfer, HeidelpayRe
         return $heidelpayRequestTransfer;
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\HeidelpayRequestTransfer $heidelpayRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\HeidelpayRequestTransfer
+     */
+    protected function mapCustoemrInformation(QuoteTransfer $quoteTransfer, HeidelpayRequestTransfer $heidelpayRequestTransfer): HeidelpayRequestTransfer
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php]: `mapCustomerInformation`
And you forgot to add return statement

File: src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php [hunk 7]
```diff
@@ -85,6 +88,47 @@ protected function mapQuoteInformation(QuoteTransfer $quoteTransfer, HeidelpayRe
...
+    }
+
+    /**
+     * @param string $date
+     *
+     * @return string
+     */
+    protected function formatDate(string $date): string
+    {
+        return (new DateTime($date))->format('Y-m-d');
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Mapper/QuoteToHeidelpayRequest.php]: It seems like it'll fail. You don't have it in use statements and PHP will try to find it under the current namespace

**Pull request merged**